src/core/account.ts

```diff
@@ -73,54 +84,108 @@ export class Account {
   /**
    * Derives an account with random private key and address
    *
-   * @param scheme optional SigningScheme - type of SigningScheme to use. Default to Ed25519
+   * @param args.scheme optional. SigningScheme - type of SigningScheme to use. Default to Ed25519
    * Currently only Ed25519 and Secp256k1 are supported
+   * @param args.legacy optional. If set to true, would create a legacy Ed25519 signing keys. Default
+   * is Single Sender signing key
    *
    * @returns Account with the given signing scheme
    */
-  static generate(scheme?: SigningScheme): Account {
+  static generate(args?: { scheme?: SigningSchemeInput; legacy?: boolean }): Account {
```

Comment:
Hmm, why is it SigningSchemeInput and not SigningScheme, is SigningScheme something else now?

Reply:
`SigningSchemeInput` is for the user to provide the signing scheme they want to use. i.e `ED25519` , `Secp256k1`, `MultiED25519`, etc.

The inner `SigningScheme` now uses the new flow `SingleSender` and later `MultiSender` so we can create it based on that.